app/src/main/java/org/mozilla/fenix/home/HomeFragment.kt

```diff
@@ -200,24 +196,31 @@ class HomeFragment : Fragment() {
             )
         )
 
-        sessionControlView = SessionControlView(homeFragmentStore, view.homeLayout, sessionControlInteractor)
-
-        ConstraintSet().apply {
-            clone(view.homeLayout)
-            connect(sessionControlView.view.id, TOP, view.wordmark.id, BOTTOM)
-            connect(sessionControlView.view.id, START, PARENT_ID, START)
-            connect(sessionControlView.view.id, END, PARENT_ID, END)
-            connect(sessionControlView.view.id, BOTTOM, view.bottom_bar.id, TOP)
-            applyTo(view.homeLayout)
-        }
+        view.homeAppBar.addOnOffsetChangedListener(
+            AppBarLayout.OnOffsetChangedListener { appBarLayout, verticalOffset ->
+                appBarLayout.alpha =
+                    1.0f - abs(verticalOffset / appBarLayout.totalScrollRange.toFloat())
+            }
+        )
 
-        @Suppress("MagicNumber") // we need constants if we define layouts in code.
-        sessionControlView.view.updateLayoutParams<ViewGroup.MarginLayoutParams> {
-            topMargin = 32.dpToPx(resources.displayMetrics)
+        if (::sessionControlView.isInitialized) {
+            if (offSet < 0) {
+                (view.homeAppBar.layoutParams as CoordinatorLayout.LayoutParams)
+                    .behavior = AppBarLayout.Behavior()
+                val behavior = ((view.homeAppBar.layoutParams as CoordinatorLayout.LayoutParams)
+                    .behavior as AppBarLayout.Behavior)
+                behavior.topAndBottomOffset = offSet
+                behavior.onNestedPreScroll(view as CoordinatorLayout, view.homeAppBar,
+                    sessionControlView.view, 0, 1, intArrayOf(2),
+                    ViewCompat.TYPE_NON_TOUCH)
+            } else {
+                view.homeAppBar.setExpanded(false)
+            }
         }
 
+        sessionControlView = SessionControlView(homeFragmentStore,
```

Comment:
This is getting set in `onCreateView` so it should presumably be destroyed in `onDestroyView`. Unfortunately, that breaks the isInitialized check above.

Maybe instead we should save the offset in `onSaveInstanceState`, in the bundle, and unpack it in `onCreateView`?